Sources/core/objects/Revolver.cpp

```diff
@@ -0,0 +1,67 @@
+#include "Revolver.h"
+
+Revolver::Revolver(size_t id) : Item(id)
+{
+    SetSprite("icons/guns.dmi");
+    SetState("revolver");
+
+    name = "Revolver",
+    
+    ammunition_ = 6;
+}
+
+bool Revolver::Handle_Ammo()
+{
+    if (ammunition_ <= 0)
+    {
+        ammunition_ = 0;
+        return false;
+    }
+    --ammunition_;
+    return true;
+}
+
+void Revolver::Shoot(sf::Vector2i target)
+{	  
+    if(Handle_Ammo())
+    {
+        id_ptr_on<Projectile> p = Create<Projectile>(Bullet::T_ITEM_S(),GetOwner());
```

Comment:
Well I knew cantouch() I looked it before , I meant how does the system know which thing you click on(I know it must be in the message which gets processed), and the other thing that for the range indication I used cube_tile->posx() and posy which returns the absolute position of tile(?). Am I right?